frontend/src/features/properties/selector/utils.ts

```diff
@@ -48,7 +50,30 @@ export const getResearchPropertyName = (
     return { label: NameSourceType.RESEARCH, value: researchProperty.propertyName };
   } else if (researchProperty.property !== undefined) {
     const property = researchProperty.property;
-    let mapProperty: IMapProperty = {
+    const mapProperty: IMapProperty = {
+      pin: property.pin?.toString(),
+      pid: property.pid?.toString(),
+      latitude: property.latitude,
+      longitude: property.longitude,
+      planNumber: property.planNumber,
+    };
+    return getPropertyName(mapProperty);
+  }
+  return { label: NameSourceType.NONE, value: '' };
+};
+
+export const getAcquisitionPropertyName = (
```

Comment:
An alternative would be to use Api_PropertyFile. I don't see any properties that are being in use here that aren't in that base interface. Would save duplicating the function once per filetype.

Reply:
I thought about that. Can try re-using but didn't want to refactor/change a lot of unrelated code. The other function seemed specific to Research file - specially setting the "source" to be research file. Don't know the code well enough to determine if changing that source to "File" would not break other forms. You still want me to look into refactoring that?

Reply:
I could go either way. If this is what you meant by setting the source to the research file:

{ label: NameSourceType.RESEARCH, value: researchProperty.propertyName };
I think that is just the property name which all fileProperties have.

Reply:
Leaving as-is until Devin's refactoring is merged. Then we can proceed.